### PR TITLE
support local secrets, passed directly to the decorators

### PR DESCRIPTION
**Pull request by iMoses**
Following #51

### CODE REVIEW

File: plugin.js [hunk 1]
```diff
@@ -47,17 +47,17 @@ function onReqHandlerWrapper (options) {
 }
 
 function plugin (fastify, options, next) {
-  const secret = options ? options.secret || '' : ''
+  const pluginSecret = options ? options.secret || '' : ''
 
   fastify.decorateRequest('cookies', {})
   fastify.decorateReply('setCookie', function setCookieWrapper (name, value, options) {
-    return fastifyCookieSetCookie(this, name, value, options, secret)
+    return fastifyCookieSetCookie(this, name, value, options, pluginSecret)
```
Comment (Eomm) [inline plugin.js]: I would add the same logic `secret || pluginSecret` to keep the same feature as per unsign

File: test/cookie.test.js [hunk 2]
```diff
@@ -273,6 +273,42 @@ test('cookies signature', (t) => {
...
 
+test('cookies local signature', (t) => {
+  t.plan(6)
+  const fastify = Fastify()
+  const secret = 'bar'
+  fastify.register(plugin)
+
+  fastify.get('/test1', (req, reply) => {
+    reply
+      .setCookie('foo', 'foo', { signed: true, secret })
```
Comment (Eomm) [inline test/cookie.test.js]: I would add a more complex test:

```suggestion
      .setCookie('foo', 'foo', { signed: true, secret })
      .setCookie('bar', 'bar', { signed: true, anotherSecret })
```

File: plugin.d.ts [hunk 3]
```diff
@@ -31,6 +31,7 @@ declare module 'fastify' {
     sameSite?: boolean | 'lax' | 'strict';
     secure?: boolean;
     signed?: boolean;
+    secret?: string;
```
Comment (Eomm) [inline plugin.d.ts]: I'm not sure this is correct, it should be added to `unsignCookie`.. but it is missing from this file!

Moreover, we need to update [this](https://github.com/fastify/fastify-cookie/blob/master/test/types.test.ts) accordingly

**Comment by mcollina** [comment]
Any updated here?

**Pull request closed without merging**